### PR TITLE
Add commons libraries

### PR DESCRIPTION
**Pull request by Azurelol**
## Description

Integrates the commons libraries to remove boilerplate code across the board.

## Changes
* ![ADD] gradle-commons-test library
* ![ADD] gradle-commons library

[NEW]:      https://resources.atlas.wooga.com/icons/icon_new.svg "New"
[ADD]:      https://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:  https://resources.atlas.wooga.com/icons/icon_improve.svg "Improve"
[CHANGE]:   https://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:      https://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:   https://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:    https://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:   https://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:      https://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:  https://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:    https://resources.atlas.wooga.com/icons/icon_webGL.svg "WebGL"
[GRADLE]:   https://resources.atlas.wooga.com/icons/icon_gradle.svg "GRADLE"
[UNITY]:    https://resources.atlas.wooga.com/icons/icon_unity.svg "Unity"
[LINUX]:    https://resources.atlas.wooga.com/icons/icon_linux.svg "Linux"
[WIN]:      https://resources.atlas.wooga.com/icons/icon_windows.svg "Windows"
[MACOS]:    https://resources.atlas.wooga.com/icons/icon_iOS.svg "macOS"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy [hunk 1]
```diff
@@ -17,49 +17,53 @@
...
  */
 class GithubBasePluginConvention {
+
     /**
      * Gradle property name to set the default value for {@code username}.
      * @value "github.username"
      * @see GithubSpec#getUsername()
      */
-    static final String GITHUB_USER_NAME_OPTION = "github.username"
+    static final PropertyLookup userName = new PropertyLookup(null, "github.username", null)
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy]: You need to keep the old fields since removing them is an API break.

**Review by Larusso — CHANGES_REQUESTED** [review]
From the looks it looks ok. But we need the constants for the option fields back in the conventions class.

File: src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy (hunk 1 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy]: Ah, my apologies.

Comment (Larusso) [reply src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy]: You could use the static Fields so you don't repeat the values. But that doesn't really matter

**Review by Larusso — APPROVED** [review]

File: src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy (hunk 1 above)
Comment (Azurelol) [reply src/main/groovy/wooga/gradle/github/base/GithubBasePluginConvention.groovy]: I can do that much.

**Pull request merged**